src/ApiConnectors/BaseApiConnector.php

```diff
@@ -17,34 +17,36 @@ abstract class BaseApiConnector implements LoggerAwareInterface
     use LoggerAwareTrait;
 
     /**
-     * Make sure to only add error messages for failure cases that caused the server not to accept / receive the
-     * request. Else the automatic retry will cause the request to be understood by the server twice.
-     *
-     * @var string[]
+     * @var AuthenticatedConnection
      */
-    private const RETRY_REQUEST_EXCEPTION_MESSAGES = [
-        "SSL: Connection reset by peer",
-        "Your logon credentials are not valid anymore. Try to log on again."
-    ];
+    private $connection;
 
     /**
      * @var int
      */
-    private const MAX_RETRIES = 3;
+    private $numRetries = 0;
 
     /**
-     * @var AuthenticatedConnection
+     * @var ApiOptions
      */
-    private $connection;
+    private $options;
 
     /**
-     * @var int
+     * @throws \InvalidArgumentException
      */
-    private $numRetries = 0;
-
-    public function __construct(AuthenticatedConnection $connection)
+    public function __construct(AuthenticatedConnection $connection, ApiOptions $options = null)
```

Comment:
PHP allows both, but I think the type should be `?ApiOptions`, to be explicit about the nullability of it.